x/datapool/keeper/pool_distribute.go

```diff
@@ -216,12 +216,60 @@ func (k Keeper) executeRevenueDistribute(ctx sdk.Context, pool *types.Pool) erro
 
 		k.SetSalesHistory(ctx, history)
 	}
+
+	err = k.sendCommissionToCurator(ctx, pool)
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
 
 func (k Keeper) getEachDistributionAmount(pool *types.Pool) sdk.Int {
-	nftPriceAmount := pool.PoolParams.NftPrice.Amount
-	numIssuedNFTs := sdk.NewIntFromUint64(pool.NumIssuedNfts)
-	targetNumData := sdk.NewIntFromUint64(pool.PoolParams.TargetNumData)
-	return nftPriceAmount.Mul(numIssuedNFTs).Quo(targetNumData)
+	nftPriceAmount := pool.PoolParams.NftPrice.Amount.ToDec()
+	numIssuedNFTs := sdk.NewIntFromUint64(pool.NumIssuedNfts).ToDec()
+	targetNumData := sdk.NewIntFromUint64(pool.PoolParams.TargetNumData).ToDec()
+	curatorCommissionRate := pool.PoolParams.CuratorCommissionRate
+
+	// ((nftPriceAmount * numIssuedNFTs) / targetNumData) * (1 - curatorCommissionRate)
+	return nftPriceAmount.Mul(numIssuedNFTs).Quo(targetNumData).Mul(sdk.NewDec(1).Sub(curatorCommissionRate)).TruncateInt()
+}
+
+// sendCommissionToCurator sends a commission to the curator
+func (k Keeper) sendCommissionToCurator(ctx sdk.Context, pool *types.Pool) error {
```

Comment:
This comment is might be optimization stuffs. 
If the above `getEachDistributionAmount()` returns both amount for each seller and curator, the calculation can be done once. But it's optional for now I think.

Reply:
It seems to good.
I applied this code.
https://github.com/medibloc/panacea-core/pull/320/commits/f838b94dbef92b29dbf2025803ac87a033750530